source/libs/cache.ts

```diff
@@ -1,26 +1,28 @@
-type CacheGetter = () => string|Promise<string>;
+ type CacheGetter<TValue extends any = any> = () => TValue | Promise<TValue>;
 
 interface CacheRequest {
 	code: string;
 	key: string;
-	value?: string | number;
+	value?: unknown;
 	expiration?: number;
 }
 
-export async function getSet(key: string, getter: CacheGetter, expiration?: number): Promise<any> {
-	const cache = await get(key);
+export async function getSet<TValue extends any = any>(key: string, getter: CacheGetter<TValue>, expiration?: number): Promise<TValue | undefined> {
+	const cache = await get<TValue>(key);
 	if (cache !== undefined) {
 		return cache;
 	}
 
 	const value = await getter();
-	if (value !== undefined) {
-		await set(key, value, expiration);
-		return value;
+	if (value === undefined) {
```

Comment:
TypeScript complains if I don't do an explicit return for undefined. We could turn the rule off for this function if you want to get rid of the explicit return.